service/history/replication/executable_task.go

```diff
@@ -355,35 +364,45 @@ func (e *ExecutableTaskImpl) DeleteWorkflow(
 }
 
 func (e *ExecutableTaskImpl) GetNamespaceInfo(
+	ctx context.Context,
 	namespaceID string,
 ) (string, bool, error) {
 	namespaceEntry, err := e.NamespaceCache.GetNamespaceByID(namespace.ID(namespaceID))
 	switch err.(type) {
 	case nil:
-		e.namespace.Store(namespaceEntry.Name())
-		shouldProcessTask := false
-	FilterLoop:
-		for _, targetCluster := range namespaceEntry.ClusterNames() {
-			if e.ClusterMetadata.GetCurrentClusterName() == targetCluster {
-				shouldProcessTask = true
-				break FilterLoop
-			}
-		}
-		return string(namespaceEntry.Name()), shouldProcessTask, nil
 	case *serviceerror.NamespaceNotFound:
-		return "", false, nil
+		if !e.ProcessToolBox.Config.EnableReplicationEagerRefreshNamespace() {
+			return "", false, nil
+		}
+
+		namespaceEntry, err = e.ProcessToolBox.EagerNamespaceRefresher.SyncNamespaceFromSourceCluster(ctx, namespace.ID(namespaceID), e.sourceClusterName)
+		if err != nil {
+			e.Logger.Info("Failed to SyncNamespaceFromSourceCluster", tag.Error(err))
```

Comment:
This is not a critical function and if it goes err, it dose not cause big impact on our system. Using info is to avoid noise error log when handling the incident.